Statistics/Distributions/Normal.cs

```diff
@@ -96,42 +91,71 @@ public IMessageLevels Validate(Utilities.IValidator<Normal> validator, out IEnum
         }
         private IRange<double> FiniteRange(double min, double max)
         {
-            double pmin = 0, epsilon = 1 / 1000000000d;
+            double pmin = 0;
             double pmax = 1 - pmin;
-            if (min.IsFinite() || max.IsFinite())//not entirely sure how inclusive or works with one sided truncation and the while loop below.
-            {
-                pmin = _Distribution.CumulativeDistribution(min);
-                pmax = _Distribution.CumulativeDistribution(max);
-            }
-            while (!(min.IsFinite() && max.IsFinite()))
+            if (min.IsFinite() || max.IsFinite())
             {
-                pmin += epsilon;
-                pmax -= epsilon;
-                if (!min.IsFinite()) min = _Distribution.InverseCumulativeDistribution(pmin);
-                if (!max.IsFinite()) max = _Distribution.InverseCumulativeDistribution(pmax);
+                pmin = CDF(min);
+                pmax = CDF(max);
             }
             return IRangeFactory.Factory(pmin, pmax);
         }
         
         #region IDistribution Functions
-        public double PDF(double x) => _Distribution.Density(x);
-        public double CDF(double x) => _Distribution.CumulativeDistribution(x);
+        public double PDF(double x){
+            return Math.Exp(-(x-Mean)*(x-Mean)/(2.0*StandardDeviation*StandardDeviation))/(Math.Sqrt(2.0*Math.PI)*StandardDeviation);
+        }
+        public double CDF(double x){
+            if (x == Double.PositiveInfinity){
+			    return 1.0;
+            }
+            else if (x == Double.NegativeInfinity){
+                return 0.0;
+            }
+            else if(x>=Mean){
+                return 0.5*(1.0+SpecialFunctions.regIncompleteGamma(0.5, (x-Mean)*(x-Mean)/(2.0*StandardDeviation*StandardDeviation)));
```

Comment:
could this logic be encapsulated in another method for readability? I am thinking a method with a name that indicates why you use the regIncompleteGamma here